EnglishLanguageArts/PartsOfSpeech/parts-of-speech.ipynb

```diff
@@ -0,0 +1,249 @@
+    "\n",
+    "import pandas as pd\n",
+    "frankenstein = pd.DataFrame() # create an empty data frame that we will append to\n",
+    "for chapter in text.split('Chapter'):\n",
+    "    if len(chapter)>500: # so that we are getting actual book chapters\n",
+    "        chapter = chapter.replace('\\r',' ').replace('\\n',' ').replace('  ', ' ') # delete the 'new line' characters\n",
+    "        frankenstein = frankenstein.append({'Chapter Text':chapter, 'Length':len(chapter)}, ignore_index=True)\n",
+    "\n",
+    "!pip install spacy --user\n",
+    "!python -m spacy download en_core_web_sm\n",
+    "from IPython.display import clear_output\n",
+    "clear_output()\n",
+    "print('Running natural language processing, please wait...')\n",
+    "import en_core_web_sm\n",
+    "nlp = en_core_web_sm.load() # set up natural language processing\n",
+    "word_types = ['ADJ', 'ADV', 'VERB', 'NOUN', 'PROPN', 'PRON'] # https://spacy.io/api/annotation#pos-tagging\n",
+    "parts_of_speech_df = pd.DataFrame(columns=word_types) # create an empty dataframe with column names\n",
+    "for i in range(len(frankenstein)): # iterate through the chapters dataframe\n",
+    "    parts_of_speech_list = [] # create an empty list\n",
+    "    for token in nlp(frankenstein['Chapter Text'][i]): # loop through each token in the chapter\n",
+    "        part_of_speech = token.pos_\n",
+    "        if part_of_speech in word_types: # if it is in the list of types we made\n",
+    "            parts_of_speech_list.append(part_of_speech)\n",
+    "    chapter_length = len(frankenstein['Chapter Text'][i])\n",
+    "    word_type_count = {} # create an empty dictionary\n",
+    "    for word_type in word_types:\n",
+    "        word_type_count.update({word_type:parts_of_speech_list.count(word_type)/chapter_length}) # add to that dictionary\n",
+    "    #word_type_count.update({'Length':chapter_length})\n",
+    "    parts_of_speech_df = parts_of_speech_df.append(word_type_count, ignore_index=True) # add to dataframe\n",
+    "parts_of_speech_df['Chapter'] = parts_of_speech_df.index+1 # create a Chapter column\n",
+    "parts_of_speech_df = parts_of_speech_df.set_index('Chapter') # set the dataframe index to be that new column\n",
+    "clear_output()\n",
+    "pd.options.plotting.backend='plotly'\n",
+    "parts_of_speech_df.plot.line(title='Parts of Speech Proportions in Frankenstein by Chapter',labels=dict(value='Proportion of Total Words'))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Pronoun Usage\n",
+    "\n",
+    "We can also look at the cumulative usage of various pronouns throughout the book."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "he = []\n",
+    "she = []\n",
+    "me = []\n",
+    "I = []\n",
+    "for row_number in range(len(frankenstein)):\n",
```

Comment:
Since this takes a while to run it would be good to have a print statement like previously 
`print('Running natural language processing, please wait...')
`